config/confighttp/confighttp.go

```diff
@@ -147,6 +158,17 @@ func (hcs *HTTPClientSettings) ToClient(host component.Host, settings component.
 
 	transport.DisableKeepAlives = hcs.DisableKeepAlives
 
+	if hcs.HTTP2ReadIdleTimeout != nil && hcs.HTTP2ReadIdleTimeout.Seconds() > 0 {
```

Comment:
rather than silently discarding negative values does returning an error make more sense?

Reply:
Actually I was not thinking of negative values; I just needed to check if it's not 0, as 0 means the healthcheck is not performed so no need to execute that block.
```
        // ReadIdleTimeout is the timeout after which a health check using ping
	// frame will be carried out if no frame is received on the connection.
	// Note that a ping response will is considered a received frame, so if
	// there is no other traffic on the connection, the health check will
	// be performed every ReadIdleTimeout interval.
	// If zero, no health check is performed.
```

Reply:
If a user manually sets a valid value, as `0s` is even though it's effectively a noop, there's no need to restrict its application in config imo.

Reply:
not certain but I think the default transport already calls it via the h2_bundle since `ForceAttemptHTTP2` is true and it specifies a dialer. I think we're having a preferential discussion so it's not a big deal to me (approved and appreciate the fix).